t/api/05-machines.t

```diff
@@ -158,6 +159,69 @@ is_deeply(
 $res = $t->delete_ok("/api/v1/machines/$machine_id")->status_is(200);
 $res = $t->delete_ok("/api/v1/machines/$machine_id")->status_is(404);    #not found
 
+subtest 'trim whitespace characters' => sub {
+    $t->post_ok(
+        '/api/v1/machines',
+        form => {
+            name                => " create_with_space ",
+            backend             => " qemu ",
+            "settings[ TEST ]"  => " test value  ",
+            "settings[TEST2  ]" => " test value2  ",
+        })->status_is(200);
+    my $id = $t->tx->res->json->{id};
+    $t->get_ok("/api/v1/machines/$id")->status_is(200);
+    is_deeply(
+        $t->tx->res->json,
+        {
+            'Machines' => [
+                {
+                    'backend'  => 'qemu',
+                    'id'       => $id,
+                    'name'     => 'create_with_space',
+                    'settings' => [
+                        {
+                            'key'   => 'TEST',
+                            'value' => 'test value'
+                        },
+                        {
+                            'key'   => 'TEST2',
+                            'value' => 'test value2'
+                        }]}]
+        },
+        'trim whitespace characters when create table'
+    ) || diag explain $t->tx->res->json;
```

Comment:
This might also be nicer as `$t->json_is({...})->or(sub { diag ... })`.

Reply:
modified